pkg/authorization/api/types.go

```diff
@@ -131,6 +131,76 @@ type PolicyBinding struct {
 	RoleBindings map[string]RoleBinding `json:"roleBindings"`
 }
 
+type ResourceAccessReviewSpec struct {
+	// Verb is one of: get, list, watch, create, update, delete
+	Verb string `json:"verb"`
+	// Resource is one of the existing resource types
+	Resource string `json:"resource"`
+	// Content is the actual content of the request for create and update
+	Content kruntime.EmbeddedObject `json:"content,omitempty"`
+	// ResourceName is the name of the resource being requested for a "get" or deleted for a "delete"
+	ResourceName string `json:"resourceName,omitempty"`
+}
+
+// ResourceAccessReviewStatus describes who can perform the action
+type ResourceAccessReviewStatus struct {
+	// Users is the list of users who can perform the action
+	Users []string `json:"users"`
+	// Groups is the list of groups who can perform the action
+	Groups []string `json:"groups"`
+	// EvaluationError is optional.  It indicates why a ResourceAccessReview failed during evaluation
+	EvaluationError string `json:"evaluationError,omitempty"`
+}
+
+// ResourceAccessReview is a means to request a list of which users and groups are authorized to perform the
+// action specified by spec
+type ResourceAccessReview struct {
+	kapi.TypeMeta   `json:",inline"`
+	kapi.ObjectMeta `json:"metadata,omitempty" `
+
+	// Spec holds the information about the action
+	Spec ResourceAccessReviewSpec `json:"spec"`
```

Comment:
It's questionable whether question/answer type apis have to be of this form (spec + status)